src/frontend/src/session.rs

```diff
@@ -383,6 +383,40 @@ impl FrontendEnv {
         });
         join_handles.push(join_handle);
 
+        // Heartbeat
+        let heartbeat_worker_node_manager = worker_node_manager.clone();
+        let join_handle = tokio::spawn(async move {
+            let duration = Duration::from_secs(std::cmp::max(
+                batch_config.mask_worker_temporary_secs as u64,
+                1,
+            ));
+
+            let mut check_heartbeat_interval =
+                tokio::time::interval(core::time::Duration::from_secs(2));
+            check_heartbeat_interval
+                .set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+            check_heartbeat_interval.reset();
+            loop {
```

Comment:
Need to listen on `shutdown_senders`

Reply:
Oops... The `shutdown_senders` and `join_handers` in `FrontendEnv` were not used... Let's fix it later 😅

Reply:
It seems the frontend node process, `shutdown_senders` and those background tasks has the same lifecycle, we don't need to handle it actually🤔